proof_of_concept/asset.py

```diff
@@ -1,21 +1,48 @@
 """Classes for describing assets."""
-from typing import Any, Dict
+from typing import Any, Dict, Union
+
+from proof_of_concept.workflow import Job
+
+
+class Metadata:
+    """Stores metadata for stored assets.
+
+    Attributes:
+        job (Job): A minimal job that will generate this asset.
+        item (str): The item in the job's workflow corresponding to
+                this asset.
+
+    """
+
+    def __init__(self, job: Job, item: str) -> None:
+        """Create a Metadata object.
+
+        Args:
+            job: A minimal job that will generate this asset.
+            item: The item in the job's workflow corresponding to this
+                    asset.
+
+        """
+        self.job = job
+        self.item = item
 
 
 class Asset:
     """Asset, a representation of a computation or piece of data."""
 
-    def __init__(self, name: str, data: Any, metadata: Any):
+    def __init__(self, id: str, data: Any,
+                 metadata: Union[Metadata, None] = None):
```

Comment:
typings has an Optional[X] generic type which is defined as Union[X, None] but is more readable.